.github/workflows/build-and-deploy-notifications.yml

```diff
@@ -67,7 +67,7 @@ jobs:
       - uses: actions/upload-artifact@v3
         with:
           name: function-app
-          path: dist/ReportingDbListener
+          path: DqtNotifications/src/DqtNotifications.ReportingDbListener/
```

Comment:
As per the [docs](https://learn.microsoft.com/en-us/azure/azure-functions/functions-how-to-github-actions?tabs=dotnet) you should be publishing the output from the build process rather than all of the source code.  Once published you should be able to see at least one function in the Azure Portal by browsing to the function app and looking at the Function blade